workorders.lic

```diff
@@ -286,28 +286,63 @@ class WorkOrders
       stock_needed = 1
       order_rem(info['stock-room'], stock_needed, info['stock-number'])
     when 'You get'
-      /are (.*) parts left of the water/ =~ bput('count my water', 'are .* parts left of the water')
+      /are (.*) parts left of the water|only (.*) part left/ =~ bput('count my water', 'are .* parts left of the water')
       existing = Regexp.last_match(1).scan(/\w+/).to_a
+      /is (.*) part left of the water/ =~ bput('count my water', 'only .* part left')
+      existing = Regexp.last_match(1).scan(/\w+/).to_a    
       val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'thirty' => 30, 'fourty' => 40, 'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
-      value = existing.map { |word| val_map[word] }.inject(&:+)
+      value = water.map { |word| val_map[word] }.inject(&:+)
```

Comment:
Hello,

You are right, one version I worked and failed had water. Can change 
water to existing.

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

On 09/04/2016 10:48 PM, Robert Herbig wrote:
>
> In workorders.lic 
> <https://github.com/rpherbig/dr-scripts/pull/737#discussion_r77465894>:
>
> >        val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'thirty' => 30, 'fourty' => 40, 'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
> > -      value = existing.map { |word| val_map[word] }.inject(&:+)
> > +      value = water.map { |word| val_map[word] }.inject(&:+)
>
> Where is |water| defined?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/rpherbig/dr-scripts/pull/737/files/7aaafb3fc3d1daa463a35a97157ca897a5e00fd4#r77465894>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AU8XXCUYoLsM6-or1Fb1oobZ2OMcb3t2ks5qm4MGgaJpZM4J0roH>.
>